web3/utils/abi.py

```diff
@@ -1,21 +1,530 @@
+        return "address"
+
+    return arg.__class__.__name__
+
+
+def get_abi_element_info(
+    abi: ABI,
+    function_identifier: FunctionIdentifier,
+    *args: Optional[Sequence[Any]],
+    abi_codec: Optional[Any] = None,
+    **kwargs: Optional[Dict[str, Any]],
+) -> ABIElementInfo:
+    """
+    Information about the function ABI, selector and input arguments.
+
+    Returns the ABI which matches the provided identifier, named arguments (``args``)
+    and keyword args (``kwargs``).
+
+    :param abi: Contract ABI.
+    :type abi: `ABI`
+    :param function_identifier: Find a function ABI with matching identifier.
+    :type function_identifier: `str`
+    :param args: Find a function ABI with matching args.
+    :type args: `Optional[Sequence[Any]]`
+    :param abi_codec: Codec used for encoding and decoding. Default with \
+    `strict_bytes_type_checking` enabled.
+    :type abi_codec: `Optional[Any]`
+    :param kwargs: Find a function ABI with matching kwargs.
+    :type kwargs: `Optional[Dict[str, Any]]`
+    :return: Function information including the ABI, selector and args.
+    :rtype: `ABIElementInfo`
+
+    .. doctest::
+
+        >>> from web3.utils.abi import get_abi_element_info
+        >>> abi = [
+        ...     {
+        ...         "constant": False,
+        ...         "inputs": [
+        ...             {"name": "a", "type": "uint256"},
+        ...             {"name": "b", "type": "uint256"},
+        ...         ],
+        ...         "name": "multiply",
+        ...         "outputs": [{"name": "result", "type": "uint256"}],
+        ...         "payable": False,
+        ...         "stateMutability": "nonpayable",
+        ...         "type": "function",
+        ...     }
+        ... ]
+        >>> fn_info = get_abi_element_info(abi, "multiply", *[7, 3])
+        >>> fn_info["abi"]
+        {'constant': False, 'inputs': [{'name': 'a', 'type': 'uint256'}, {\
+'name': 'b', 'type': 'uint256'}], 'name': 'multiply', 'outputs': [{\
+'name': 'result', 'type': 'uint256'}], 'payable': False, \
+'stateMutability': 'nonpayable', 'type': 'function'}
+        >>> fn_info["selector"]
+        '0x165c4a16'
+        >>> fn_info["arguments"]
+        (7, 3)
+    """
+    fn_abi = get_abi_element(
+        abi, function_identifier, *args, abi_codec=abi_codec, **kwargs
+    )
+    fn_selector = encode_hex(function_abi_to_4byte_selector(fn_abi))
+    fn_inputs: Tuple[Any, ...] = tuple()
+
+    if fn_abi["type"] == "fallback" or fn_abi["type"] == "receive":
+        return ABIElementInfo(abi=fn_abi, selector=encode_hex(b""), arguments=tuple())
+    else:
+        fn_inputs = get_normalized_abi_inputs(fn_abi, *args, **kwargs)
+        _, aligned_fn_inputs = get_aligned_abi_inputs(fn_abi, fn_inputs)
+
+        return ABIElementInfo(
+            abi=fn_abi, selector=fn_selector, arguments=aligned_fn_inputs
+        )
+
+
+def get_abi_element(
+    abi: ABI,
+    function_identifier: FunctionIdentifier,
```

Comment:
Also here... is this just for functions or other elements (e.g. ABIEvent). It looks like it returns `ABIElement` so maybe this arg should reflect this.